File: 프로그래머스/1/172928. 공원 산책/공원 산책.java
class Solution {
    public int[] solution(String[] park, String[] routes) {
        int sx = 0, sy = 0;
        for(int i = 0; i< park.length; i++) {
            for(int j = 0; j< park[i].length(); j++) {
                if(park[i].charAt(j) == 'S') {
                    sx = i;
                    sy = j;
                }
            }
        }
        // 북남 -> sx 동서 -> sy
        System.out.print(sx + " " +sy); // 시작 위치 좌표
        for(int i = 0; i < routes.length; i++) {
            String[] str = routes[i].split(" ");
            String direction = str[0];
            int move = Integer.parseInt(str[1]);
            if(direction.equals("E") && sy + move < park[0].length()) {
                boolean isGo = true;
                for(int j = sy + 1; j <= sy + move; j++) {
                    if(park[sx].charAt(j) == 'X') {
                        isGo = false;
                        break;
                    }
                }
                if(isGo) {
                    sy += move;
                }
               
               
            }
            else if(direction.equals("W") && sy - move >= 0) {
                boolean isGo = true;
                for(int j = sy - 1; j >= sy - move; j--) {
                    if(park[sx].charAt(j) == 'X') {
                        isGo = false;
                        break;
                    }
                }
                if(isGo) {
                    sy -= move;
                }
                
            }
            else if(direction.equals("S") && sx + move < park.length) {
                boolean isGo = true;
                for(int j = sx + 1; j <= sx + move; j++) {
                    if(park[j].charAt(sy) == 'X') {
                        isGo = false;
                        break;
                    }
                }
                if(isGo) {
                    sx += move;
                }
              
            }
            else if(direction.equals("N") && sx - move >= 0) {
                boolean isGo = true;
                for(int j = sx - 1; j >= sx - move; j--) {
                    if(park[j].charAt(sy) == 'X') {
                        isGo = false;
                        break;
                    }
                }
                if(isGo) {
                    sx -= move;
                }
             
            }
            
            
        }
        int[] answer = {sx, sy};
        return answer;
    }
}

File: 프로그래머스/1/172928. 공원 산책/공원 산책.js
function solution(park, routes) {
    var answer = [];
    let pos = [0,0];
    const rows = park.length;
    const cols = park[0].length;
    for (let i = 0; i < rows; i++) {
        for (let j = 0; j < cols; j++) {
            if(park[i][j] === 'S') {
                pos = [i,j];
            }
        }
    }
    
    for (const route of routes) {
        let [direction, move] = route.split(' ');
        move = parseInt(move);
        switch(direction) { // 이동 후 좌표가 격자 안넘어가는지 이동 중 X 있는지
            case 'E':
                if (pos[1] + move >= cols) break;
                let go1 = true;
                for (let i = 1; i <= move; i++) {
                    const dx = pos[1] + i;
                    if(park[pos[0]][dx] === 'X') {
                        go1 = false;
                        break;
                    }
                }
                if (go1) pos[1] += move;
                break;
            case 'W':
                if (pos[1] - move < 0) break;
                let go2 = true;
                for (let i = 1; i <= move; i++) {
                    const dx = pos[1] - i;
                    if (park[pos[0]][dx] === 'X') {
                        go2 = false;
                        break;
                    }
                }
                if(go2) pos[1] -= move;
                break;
            case 'N':
                if (pos[0] - move < 0) break;
                let go3 = true;
                for (let i = 1; i <= move; i++) {
                    const dy = pos[0] - i;
                    if(park[dy][pos[1]] === 'X') {
                        go3 = false;
                        break;
                    }
                }
                if(go3) pos[0] -= move;
                break;
            case 'S':
                if (pos[0] + move >= cols) break;
                let go4 = true;
                for (let i = 1; i <= move; i++) {
                    const dy = pos[0] + i;
                    if (park[dy][pos[1]] === 'X') {
                        go4 = false;
                        break;
                    }
                }
                if (go4) pos[0] += move;
                break;
        }
    }
    answer = pos;
    return answer;
}